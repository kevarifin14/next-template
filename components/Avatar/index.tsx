import { sizeToDimension, classNames, TailwindSize } from 'utils/tailwind';

type AvatarProps = {
  src: string,
  size?: TailwindSize,
  className?: string,
}

export default function Avatar({ src, size = 'md', className }: AvatarProps) {
  const dimension = sizeToDimension[size];
  const sizeClassName = `h-${dimension} w-${dimension} text-${size}`;

  const avatarClassName = classNames(
    'relative', 'rounded-full', 'overflow-hidden', 'flex', 'items-center',
    sizeClassName,
    className,
  );

  return (
    <div className={avatarClassName}>
      <img src={src} alt={src} />
    </div>
  );
}
