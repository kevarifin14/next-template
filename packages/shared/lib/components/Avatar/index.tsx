import React from 'react';

import { classNames, TailwindSize, sizeToClassName } from '../../utils/tailwind';

type AvatarProps = {
  src: string,
  size?: TailwindSize,
  className?: string,
};

export function Avatar({ src, size = 'md', className }: AvatarProps) {
  const sizeClassName = sizeToClassName(size);

  const avatarClassName = classNames(
    'relative', 'rounded-full', 'overflow-hidden', 'flex', 'items-center', `text-${size}`,
    sizeClassName,
    className,
  );

  return (
    <div className={avatarClassName}>
      <img src={src} alt={src} />
    </div>
  );
}
