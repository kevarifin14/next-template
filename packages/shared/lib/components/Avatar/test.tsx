import { render } from '@testing-library/react';
import React from 'react';

import { Avatar } from '.';

describe('Avatar', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(<Avatar src="" />);
    expect(baseElement).toMatchSnapshot();
  });
});
