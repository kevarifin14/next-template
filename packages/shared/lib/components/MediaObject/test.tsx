import { render } from '@testing-library/react';
import React from 'react';

import { MediaObject } from '.';
import * as AvatarSpy from '../Avatar';

describe('MediaObject', () => {
  const mockSrc = 'mockSrc';
  const mockTitle = 'mockTitle';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    const { baseElement } = render(<MediaObject src={mockSrc} title={mockTitle} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('renders Avatar', () => {
    const avatarSpy = jest.spyOn(AvatarSpy, 'Avatar');
    render(<MediaObject src={mockSrc} title={mockTitle} size="2xl" />);
    expect(avatarSpy).toHaveBeenCalled();
  });
});
