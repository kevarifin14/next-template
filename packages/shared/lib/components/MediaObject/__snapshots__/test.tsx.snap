// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaObject matches snapshot 1`] = `
<body>
  <div>
    <div
      class="flex items-center space-x-2"
    >
      <div
        class="flex-shrink-0"
      >
        <div
          class="relative rounded-full overflow-hidden flex items-center text-md h-10 w-10"
        >
          <img
            alt="mockSrc"
            src="mockSrc"
          />
        </div>
      </div>
      <div
        class="flex flex-col justify-center overflow-hidden"
      >
        <p
          class="font-medium text-undefined truncate"
        >
          mockTitle
        </p>
        <p
          class="text-xs truncate"
        />
      </div>
    </div>
  </div>
</body>
`;
